cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -189,6 +189,10 @@ void reader_impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num_
     initial_str_offsets =
       cudf::detail::make_device_uvector_async(host_offsets_vector, _stream, _mr);
     chunk_nested_str_data.host_to_device_async(_stream);
+
+    // Allocate string offset buffers and get string offsets for non-dictionary, non-FLBA string
+    // columns
+    set_page_string_offset_indices(skip_rows, num_rows, _subpass_page_mask);
```

Comment:
Optional: Perhaps a better name for this function could be `compute_page_string_offset_indices`